### PR TITLE
BaSP-TG-158: Implemented firebase to app

### PR DESCRIPTION
**Pull request by marcomigliaro**
- Log In Screen
- Added route for log in and log out (in header)
- Created NotAllowed Screen (it isn´t routed yet)
- Implement firebase
- Added actions, constants, reducer and thunks for log in and log out

### CODE REVIEW

File: src/Components/Auth/Login/Login.jsx [hunk 1]
```diff
@@ -0,0 +1,103 @@
+import { useForm } from 'react-hook-form';
+import Input from 'Components/SharedComponents/Input/Input';
+import Modal from 'Components/SharedComponents/Modal/Modal';
+import Button from 'Components/SharedComponents/Button/Button';
+import { login } from 'Components/redux/modules/auth/thunks';
+// import { cleanError } from 'redux//modules/auth/actions';
```
Comment (SuarezAlexisEzequiel) [inline src/Components/Auth/Login/Login.jsx:6]: Can this comment be removed?

File: src/Components/Auth/Login/Login.jsx [hunk 2]
```diff
@@ -0,0 +1,103 @@
...
+    .required()
+    .regex(/^(?=.*?[a-zA-Z])(?=.*?[0-9])/)
+    .message('Invalid password, it must contain letters and numbers')
+});
+
+const Login = (props) => {
+  const [isOpen, setIsOpen] = useState(false);
+  const [role, setRole] = useState(false);
+
+  // const error = useSelector((store) => store.auth.error);
```
Comment (SuarezAlexisEzequiel) [inline src/Components/Auth/Login/Login.jsx:34]: Can this comment be removed?

File: src/Components/Auth/Login/Login.jsx [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    });
+  };
+
+  const {
+    handleSubmit,
+    register,
+    formState: { errors }
+  } = useForm({ mode: 'onChange', resolver: joiResolver(schema) });
+
+  // const required = (value) => (value ? undefined : 'Required');
```
Comment (SuarezAlexisEzequiel) [inline src/Components/Auth/Login/Login.jsx:53]: Can this comment be removed?

File: src/Components/Auth/Login/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      case 'EMPLOYEE':
+        return props.history.push('/employee/projects');
+      case 'ADMIN':
+        return props.history.push('/admins');
+      case 'SUPERADMIN':
+        return props.history.push('/super-admins');
+      default:
+        break;
+    }
+    // callback: () => dispatch(cleanError());
```
Comment (SuarezAlexisEzequiel) [inline src/Components/Auth/Login/Login.jsx:66]: Can this comment be removed?

File: src/Components/helper/firebase/index.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+import firebase from 'firebase/compat/app';
+import 'firebase/compat/auth';
+// import { store } from 'Components/redux/store';
```
Comment (SuarezAlexisEzequiel) [inline src/Components/helper/firebase/index.js:3]: Can this comment be removed?

**Review by SuarezAlexisEzequiel — COMMENTED** [review]
Can comments be removed?

**Review by gregorioelias — APPROVED** [review]

**Review by Juanoauil — APPROVED** [review]

**Review by Lucas-DeLorenzi — APPROVED** [review]

**Review by JulianDeme — APPROVED** [review]

**Pull request merged**